src/Modules/UngradedModules/SubModuleOfFreeModule.jl

```diff
@@ -476,10 +476,38 @@ end
 Check if `a` is an element of `M`.
 """
 function in(a::FreeModElem, M::SubModuleOfFreeModule)
-  F = ambient_free_module(M)
-  return iszero(reduce(a, standard_basis(M, ordering=default_ordering(F))))
+    iszero(a) && return true
+    any(==(a), gens(M)) && return true
+    return in_atomic(a, M)
+end
+
+function in_atomic(a::FreeModElem{T}, M::SubModuleOfFreeModule) where {S<:Union{ZZRingElem,<:FieldElem}, T<:MPolyRingElem{S}}
+    F = ambient_free_module(M)
+    return iszero(reduce(a, standard_basis(M, ordering=default_ordering(F))))
+end
+
+@attr Any function solve_ctx(M::SubModuleOfFreeModule)
```

Comment:
We might be able to get eventually rid of the `Any` here and replace it by something concrete, see https://github.com/Nemocas/AbstractAlgebra.jl/pull/2101 for a proposal for a programmatic way to compute the type of solve context that can then replace the `Any`